File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.py
def solution(today, terms, privacies):
    answer = []
    ty, tm, td = map(int, today.split('.'))
    expiration = {}

    for term in terms:
        t, m = term.split()
        expiration[t] = int(m) * 28

    for i in range(len(privacies)):
        date, type = privacies[i].split()
        dy, dm, dd = map(int, date.split('.'))
        y, m, d = ty - dy, tm - dm, td - dd

        total = y * 336 + m * 28 + d
        if expiration[type] <= total:
            answer.append(i+1)

    return answer

File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.js
function solution(today, terms, privacies) {
  let result = [];
  expiration = {};

  const [tY, tM, tD] = today.split(".");
  terms.forEach(term => {
    const [type, t] = term.split(" ");
    expiration[type] = t * 28;
  })

  privacies.forEach((privacy, idx) => {
    const [dates, type] = privacy.split(" ");
    const [pY, pM, pD] = dates.split(".");
    
    const [y, m, d] = [tY - pY, tM - pM, tD - pD];
    const total = 336*y + 28*m + d;
    if(expiration[type] <= total) result.push(idx+1);
  })

  return result;
}